### PR TITLE
Fixes after the PR "CGAL I/O

### PR DESCRIPTION
**Pull request by lrineau**
## Summary of Changes

Fix errors made in PR https://github.com/CGAL/cgal/pull/4255. A lot of deprecated functions were no longer tested, and wrongly implemented.

## Release Management

* Affected package(s): BGL Point_set_3 Point_set_processing_3 Stream_support  Surface_mesh

### CODE REVIEW

**Comment by lrineau** [comment]
@maxGimeno Make sure all deprecated function is actually tested, and amend this PR.

**Comment by sloriot** [comment]
One test that could be done is to try to compile CGAL examples from 5.2 with master.

File: BGL/include/CGAL/boost/graph/IO/GOCAD.h [hunk 1]
```diff
@@ -303,8 +303,12 @@ bool write_GOCAD(std::ostream& os,
...
     return false;
 
-  const int precision = choose_parameter(get_parameter(np, internal_np::stream_precision), 6);
-  os.precision(precision);
+  if(!parameters::is_default_parameter(
+       parameters::get_parameter(np, internal_np::stream_precision)))
+  {
+    const int precision = choose_parameter(get_parameter(np, internal_np::stream_precision), 6);
+    os.precision(precision);
+  }
```
Comment (lrineau) [inline BGL/include/CGAL/boost/graph/IO/GOCAD.h:311]: I see those lines copy-pasted several times... Should not it be in a function?

**Comment by lrineau** [comment]
Compilation errors in Github actions checks: https://github.com/CGAL/cgal/pull/5352/checks?check_run_id=1813116033#step:4:4581
> /home/runner/work/cgal/cgal/Stream_support/include/CGAL/IO/OI/Inventor_ostream.h: In function ‘void CGAL::set_default_stream_precision(CGAL::Inventor_ostream_base&, const NP&)’:
/home/runner/work/cgal/cgal/Stream_support/include/CGAL/IO/OI/Inventor_ostream.h:95:7: error: ‘parameters’ has not been declared
   if(!parameters::is_default_parameter(
       ^~~~~~~~~~

**Comment by maxGimeno** [comment]
fixed

**Comment by maxGimeno** [comment]
Success in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-55.shtml

File: BGL/include/CGAL/boost/graph/IO/GOCAD.h [hunk 2]
```diff
@@ -271,7 +271,7 @@ bool read_GOCAD(const std::string& fname, Graph& g,
 ///   \cgalParamNBegin{stream_precision}
 ///     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
 ///     \cgalParamType{int}
-///     \cgalParamDefault{`6`}
+///     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/GOCAD.h:274]:
```suggestion
///     \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/GOCAD.h [hunk 3]
```diff
@@ -454,6 +453,8 @@ bool write_GOCAD(const std::string& fname,
 {
   std::ofstream os(fname);
   CGAL::set_mode(os, CGAL::IO::ASCII);
+
+
   return write_GOCAD(os, fname.c_str(), g, np);
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/GOCAD.h:458]:
```suggestion

  return write_GOCAD(os, fname.c_str(), g, np);
```

File: BGL/include/CGAL/boost/graph/IO/GOCAD.h [hunk 4]
```diff
@@ -380,7 +379,7 @@ bool write_GOCAD(std::ostream& os, const char* name, const Graph& g,
 ///   \cgalParamNBegin{stream_precision}
 ///     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
 ///     \cgalParamType{int}
-///     \cgalParamDefault{`6`}
+///     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/GOCAD.h:382]:
```suggestion
///     \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/OBJ.h [hunk 5]
```diff
@@ -231,7 +231,7 @@ bool read_OBJ(const std::string& fname, Graph& g,
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/OBJ.h:234]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/OFF.h [hunk 6]
```diff
@@ -300,6 +300,19 @@ CGAL_DEPRECATED bool read_off(const char* fname, Graph& g, const CGAL_BGL_NP_CLA
  return read_OFF(fname, g, np);
 }
 
+
+template <typename Graph>
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/OFF.h:304]:
```suggestion
template <typename Graph>
```

File: BGL/include/CGAL/boost/graph/IO/OFF.h [hunk 7]
```diff
@@ -374,7 +387,7 @@ bool write_OFF_BGL(std::ostream& os,
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/OFF.h:390]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/PLY.h [hunk 8]
```diff
@@ -311,7 +311,7 @@ bool read_PLY(const std::string& fname, Graph& g,
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/PLY.h:314]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/STL.h [hunk 9]
```diff
@@ -231,7 +231,7 @@ bool read_STL(const std::string& fname, Graph& g) { return read_STL(fname, g, pa
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
        \cgalParamType{int}
-       \cgalParamDefault{`6`}
+       \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/STL.h:234]:
```suggestion
       \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/VTK.h [hunk 10]
```diff
@@ -420,7 +420,7 @@ void write_polys_points(std::ostream& os,
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/VTK.h:423]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/IO/WRL.h [hunk 11]
```diff
@@ -59,7 +59,7 @@ namespace CGAL {
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/IO/WRL.h:62]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: BGL/include/CGAL/boost/graph/named_params_helper.h [hunk 12]
```diff
@@ -560,6 +560,18 @@ CGAL_DEF_GET_INITIALIZED_INDEX_MAP(face, typename boost::graph_traits<Graph>::fa
     Alpha_expansion_boost_adjacency_list_tag
     >::type type;
   };
+
+  template<typename NP>
+  void set_default_stream_precision(std::ostream& os, const NP& np)
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/named_params_helper.h:565]: Why "default"? It's more like "set_stream_precision_from_NP".

File: BGL/include/CGAL/boost/graph/named_params_helper.h [hunk 13]
```diff
@@ -560,6 +560,18 @@ CGAL_DEF_GET_INITIALIZED_INDEX_MAP(face, typename boost::graph_traits<Graph>::fa
...
+  {
+    if(!parameters::is_default_parameter(
+         parameters::get_parameter(np, internal_np::stream_precision)))
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/named_params_helper.h:568]:
```suggestion
    if(!parameters::is_default_parameter(get_parameter(np, internal_np::stream_precision)))
```

File: BGL/include/CGAL/boost/graph/named_params_helper.h [hunk 14]
```diff
@@ -560,6 +560,18 @@ CGAL_DEF_GET_INITIALIZED_INDEX_MAP(face, typename boost::graph_traits<Graph>::fa
...
+    {
+      const int precision = parameters::choose_parameter<int>(
+            parameters::get_parameter(np, internal_np::stream_precision));
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/named_params_helper.h:571]:
```suggestion
      const int precision = parameters::choose_parameter<int>(get_parameter(np, internal_np::stream_precision));
```

File: Maintenance/infrastructure/cgal.geometryfactory.com/crontab [hunk 15]
```diff
@@ -67,7 +67,7 @@ LC_CTYPE=en_US.UTF-8
 18 * * * * $HOME/bin/dump_crontab
 
 # Docker check every hour
-0 * * * * docker inspect --format='{{json .State.Health.Status}}' cgal-mediawiki-docker_wiki_1 | grep -q '"healthy"' || docker logs cgal-mediawiki-docker_wiki_1
+#0 * * * * docker inspect --format='{{json .State.Health.Status}}' cgal-mediawiki-docker_wiki_1 | grep -q '"healthy"' || docker logs cgal-mediawiki-docker_wiki_1
```
Comment (MaelRL) [inline Maintenance/infrastructure/cgal.geometryfactory.com/crontab:70]: Alien change

File: Point_set_3/include/CGAL/Point_set_3/IO/OFF.h [hunk 16]
```diff
@@ -134,7 +134,7 @@ CGAL_DEPRECATED bool read_off_point_set(std::istream& is,  ///< input stream.
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_3/include/CGAL/Point_set_3/IO/OFF.h:137]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_3/include/CGAL/Point_set_3/IO/PLY.h [hunk 17]
```diff
@@ -456,7 +456,7 @@ CGAL_DEPRECATED bool read_ply_point_set(std::istream& is, ///< input stream.
    \cgalParamNBegin{stream_precision}
      \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
      \cgalParamType{int}
-     \cgalParamDefault{`6`}
+     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_3/include/CGAL/Point_set_3/IO/PLY.h:459]:
```suggestion
     \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_3/include/CGAL/Point_set_3/IO/XYZ.h [hunk 18]
```diff
@@ -132,7 +132,7 @@ CGAL_DEPRECATED bool read_xyz_point_set(std::istream& is, CGAL::Point_set_3<Poin
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_3/include/CGAL/Point_set_3/IO/XYZ.h:135]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_processing_3/include/CGAL/IO/write_off_points.h [hunk 19]
```diff
@@ -125,7 +124,7 @@ bool write_OFF_PSP(std::ostream& os,
     \cgalParamNBegin{stream_precision}
       \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
       \cgalParamType{int}
-      \cgalParamDefault{`6`}
+      \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_processing_3/include/CGAL/IO/write_off_points.h:127]:
```suggestion
      \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_processing_3/include/CGAL/IO/write_ply_points.h [hunk 20]
```diff
@@ -175,7 +175,7 @@ template <typename PointRange,
      \cgalParamNBegin{stream_precision}
        \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
        \cgalParamType{int}
-       \cgalParamDefault{`6`}
+       \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_processing_3/include/CGAL/IO/write_ply_points.h:178]:
```suggestion
       \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_processing_3/include/CGAL/IO/write_xyz_points.h [hunk 21]
```diff
@@ -122,7 +121,7 @@ bool write_XYZ_PSP(std::ostream& os,
      \cgalParamNBegin{stream_precision}
        \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
        \cgalParamType{int}
-       \cgalParamDefault{`6`}
+       \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Point_set_processing_3/include/CGAL/IO/write_xyz_points.h:124]:
```suggestion
       \cgalParamDefault{the precision of the stream `os`}
```

File: Point_set_processing_3/include/CGAL/IO/write_xyz_points.h [hunk 22]
```diff
@@ -202,6 +201,7 @@ bool write_XYZ(const std::string& filename,
                )
 {
   std::ofstream os(filename);
+
```
Comment (MaelRL) [inline Point_set_processing_3/include/CGAL/IO/write_xyz_points.h:204]:
```suggestion
```

File: Stream_support/include/CGAL/IO/GOCAD.h [hunk 23]
```diff
@@ -357,7 +356,7 @@ bool write_GOCAD(std::ostream& os,
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/GOCAD.h:359]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Stream_support/include/CGAL/IO/OBJ.h [hunk 24]
```diff
@@ -318,7 +319,7 @@ bool read_OBJ(const std::string& fname, PointRange& points, PolygonRange& polygo
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/OBJ.h:322]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Stream_support/include/CGAL/IO/OFF.h [hunk 25]
```diff
@@ -298,7 +298,7 @@ bool read_OFF(const std::string& fname, PointRange& points, PolygonRange& polygo
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/OFF.h:301]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Stream_support/include/CGAL/IO/OFF.h [hunk 26]
```diff
@@ -367,6 +367,7 @@ bool write_OFF(const std::string& fname,
                )
 {
   std::ofstream os(fname);
+
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/OFF.h:370]:
```suggestion
```

File: Stream_support/include/CGAL/IO/OI/Inventor_ostream.h [hunk 27]
```diff
@@ -89,6 +90,18 @@ class Inventor_ostream
   }
 };
 
+template<typename NP>
+void set_default_stream_precision(Inventor_ostream_base& os, const NP& np)
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/OI/Inventor_ostream.h:94]: just call the other function with os.os()?

File: Stream_support/include/CGAL/IO/PLY.h [hunk 28]
```diff
@@ -444,7 +445,7 @@ bool read_PLY(const std::string& fname, PointRange& points, PolygonRange& polygo
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/PLY.h:448]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Stream_support/include/CGAL/IO/STL.h [hunk 29]
```diff
@@ -284,7 +284,7 @@ bool read_STL(const std::string& fname, PointRange& points, TriangleRange& facet
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/STL.h:287]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Stream_support/include/CGAL/IO/VRML/File_writer_VRML_2.h [hunk 30]
```diff
@@ -28,6 +28,7 @@ namespace CGAL {
 
 class File_writer_VRML_2
 {
+
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/VRML/File_writer_VRML_2.h:31]:
```suggestion
```

File: Stream_support/include/CGAL/IO/VRML/File_writer_VRML_2.h [hunk 31]
```diff
@@ -42,7 +43,7 @@ class File_writer_VRML_2
                     std::size_t facets,
                     const bool /*colors*/ = false,
                     const bool /*normals*/ = false,
-                    const bool /*textures*/ = false)
+                    const bool /*te xtures*/ = false)
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/VRML/File_writer_VRML_2.h:46]:
```suggestion
                    const bool /*textures*/ = false)
```

File: Stream_support/include/CGAL/IO/VRML/VRML_2_ostream.h [hunk 32]
```diff
@@ -104,6 +105,17 @@ inline VRML_2_ostream& operator<<(VRML_2_ostream& os,
   return os;
 }
 
+template<typename NP>
+void set_default_stream_precision(VRML_2_ostream& os, const NP& np)
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/VRML/VRML_2_ostream.h:109]: same remark as above

File: Stream_support/include/CGAL/IO/VTK.h [hunk 33]
```diff
@@ -19,6 +19,8 @@
 #include <CGAL/IO/VTK/VTK_writer.h>
 
 #include <CGAL/boost/graph/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/VTK.h:23]:
```suggestion
```

File: Stream_support/include/CGAL/IO/VTK.h [hunk 34]
```diff
@@ -373,7 +375,7 @@ void write_soup_polys_points(std::ostream& os,
  *   \cgalParamNBegin{stream_precision}
  *     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
  *     \cgalParamType{int}
- *     \cgalParamDefault{`6`}
+ *     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Stream_support/include/CGAL/IO/VTK.h:378]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Surface_mesh/include/CGAL/Surface_mesh/IO/OFF.h [hunk 35]
```diff
@@ -556,7 +556,7 @@ bool write_OFF_with_or_without_vnormals(std::ostream& os,
 ///   \cgalParamNBegin{stream_precision}
 ///     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
 ///     \cgalParamType{int}
-///     \cgalParamDefault{`6`}
+///     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Surface_mesh/include/CGAL/Surface_mesh/IO/OFF.h:559]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

File: Surface_mesh/include/CGAL/Surface_mesh/IO/PLY.h [hunk 36]
```diff
@@ -899,7 +899,7 @@ CGAL_DEPRECATED bool read_ply(std::istream& is, Surface_mesh<P>& sm, std::string
 ///   \cgalParamNBegin{stream_precision}
 ///     \cgalParamDescription{a parameter used to set the precision (i.e. how many digits are generated) of the output stream}
 ///     \cgalParamType{int}
-///     \cgalParamDefault{`6`}
+///     \cgalParamDefault{`the precision of the given stream`}
```
Comment (MaelRL) [inline Surface_mesh/include/CGAL/Surface_mesh/IO/PLY.h:902]:
```suggestion
 *     \cgalParamDefault{the precision of the stream `os`}
```

**Review by MaelRL — CHANGES_REQUESTED** [review]

File: Maintenance/infrastructure/cgal.geometryfactory.com/crontab (hunk 15 above)
Comment (maxGimeno) [reply Maintenance/infrastructure/cgal.geometryfactory.com/crontab:70]: oO

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-65.shtml

**Pull request merged**